src/PrestaShopBundle/Controller/Admin/Improve/International/TaxRulesGroupController.php

```diff
@@ -104,7 +127,42 @@ public function createAction(Request $request): Response
      */
     public function editAction(Request $request, int $taxRulesGroupId): Response
     {
-        return $this->redirect($this->getAdminLink('AdminTaxRulesGroup', []));
+        try {
+            /** @var EditableTaxRulesGroup $editableTaxRulesGroup */
+            $editableTaxRulesGroup = $this->getQueryBus()->handle(new GetTaxRulesGroupForEditing((int) $taxRulesGroupId));
+        } catch (TaxRulesGroupException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_tax_rules_groups_index');
+        }
+
+        $taxRulesGroupForm = null;
+
+        try {
+            $taxRulesGroupForm = $this->getFormBuilder()->getFormFor((int) $taxRulesGroupId);
+            $taxRulesGroupForm->handleRequest($request);
+            $result = $this->getFormHandler()->handleFor((int) $taxRulesGroupId, $taxRulesGroupForm);
+            if ($result->isSubmitted() && $result->isValid()) {
+                $this->addFlash('success', $this->trans('Update successful', 'Admin.Notifications.Success'));
+
+                return $this->redirectToRoute('admin_tax_rules_groups_edit', [
+                    'taxRulesGroupId' => $taxRulesGroupId,
+                ]);
+            }
+        } catch (TaxRulesGroupNotFoundException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+
+            return $this->redirectToRoute('admin_tax_rules_groups_index');
+        } catch (Exception $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->render('@PrestaShop/Admin/Improve/International/TaxRulesGroup/edit.html.twig', [
+            'enableSidebar' => true,
+            'layoutTitle' => $this->trans('Edit: %value%', 'Admin.Actions', ['%value%' => $editableTaxRulesGroup->getName()]),
```

Comment:
I wonder if you could not get this data from the `$taxRulesGroupForm->getData()` It would avoid using the query twice because it's called here in the controller once, and then a second time via the form builder

Reply:
Fixed. Good idea :bulb:.